### DISCUSSION TITLE
A little help? Dependencies

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#4559] [Q&A] [answered]
### What is the issue?

I'm trying to figure out what is or is not a dependency for the macOS client after running into a problem with the current libunistring 1.1 from Homebrew, that was updated a couple of days ago. The build directory is emptied before each build attempt, and i'm using make rather than ninja. I _thought_ that idn2 and unistring were dependencies for the qt client - but not the macOS client. Things started with the below, but the macOS client would not launch.

```
warning: cannot resolve item '/usr/local/opt/libunistring/lib/libunistring.2.dylib'
```

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

```
CMake Error at /Users/garyelshaw/Documents/GitHub/transmission-primary/cmake/TrMacros.cmake:302 (file):
  file COPY cannot find
  "/usr/local/opt/libunistring/lib/libunistring.2.dylib": No such file or
  directory.
Call Stack (most recent call first):
  macosx/cmake_install.cmake:64 (tr_fixup_bundle_item)
  cmake_install.cmake:65 (include)


make: *** [install] Error 1
```

Okay then.

brew install libunistring@1.0 would get to the configure stage and essentially freeze. So i thought well, let's try MacPorts idn2 and unistring after uninstalling idn2 and unistring from homebrew. Nope. So i disabled the qt client to OFF in the cmake lists. Nope.

The consistency seems to be with TrMacros. Any ideas?

```
CMake Error at /Users/garyelshaw/Documents/GitHub/transmission-primary/cmake/TrMacros.cmake:302 (file):
  file COPY cannot find "/usr/local/opt/libidn2/lib/libidn2.0.dylib": No such
  file or directory.
Call Stack (most recent call first):
  macosx/cmake_install.cmake:64 (tr_fixup_bundle_item)
  cmake_install.cmake:65 (include)


make: *** [install] Error 1
```

### Which application of Transmission?

macOS and qt app

### Which version of Transmission?

Beta 3

**Comment by mikedld** [comment]
Packages installed via Homebrew aren't meant to be used for distrubition. If you want to use CMake and you want to create a redistributable application bundle (not sure why else would you want to build the `install` target), you need to disable the use of system dependencies and use bundled ones, i.e.

```
cmake ... \
    -DUSE_SYSTEM_EVENT2=OFF \
    -DUSE_SYSTEM_DEFLATE=OFF \
    -DUSE_SYSTEM_DHT=OFF \
    -DUSE_SYSTEM_MINIUPNPC=OFF \
    -DUSE_SYSTEM_NATPMP=OFF \
    -DUSE_SYSTEM_UTP=OFF \
    -DUSE_SYSTEM_B64=OFF \
    -DUSE_SYSTEM_PSL=OFF
```

This is basically what's happening when building using the Xcode project, it doesn't use any system-wide libraries except for libcurl.

If you just want to run the app from the build tree, you don't need to build the `install` target, it should work as is, even if you use some of the dependencies from the system (so, not configuring with any of the `USE_SYSTEM_*` options explicitly disabled).

**Comment by GaryElshaw** [comment, marked as answer]
Many thanks Mike, i'll give /usr/local a clean.
